drivers/display/display_sh1122.c

```diff
@@ -0,0 +1,515 @@
+/*
+ * Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sh1122, CONFIG_DISPLAY_LOG_LEVEL);
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/mipi_dbi.h>
+#include <zephyr/kernel.h>
+
+#define SH1122_CONTROL_ALL_BYTES_CMD  0x0
+#define SH1122_CONTROL_ALL_BYTES_DATA 0x40
+
+#define SH1122_SET_PHASE_LENGTH    0xD9
+#define SH1122_SET_OSC_FREQ        0xD5
+#define SH1122_SET_VCOMH           0xDB
+#define SH1122_SET_DCDC            0xAD
+#define SH1122_SET_DISPLAY_OFFSET  0xD3
+#define SH1122_DISPLAY_ON          0xAF
+#define SH1122_DISPLAY_OFF         0xAE
+#define SH1122_SET_MULTIPLEX_RATIO 0xA8
+#define SH1122_SET_SEG_ORDER_10    0xA0
+#define SH1122_SET_SEG_ORDER_01    0xA1
+#define SH1122_SET_COM_ORDER_10    0xC0
+#define SH1122_SET_COM_ORDER_01    0xC8
+#define SH1122_SET_CONTRAST_CTRL   0x81
+#define SH1122_SET_VSEGM           0xDC
+#define SH1122_SET_DISPLAY_RAM     0xA4
+#define SH1122_SET_DISPLAY_ALL_ON  0xA5
+#define SH1122_SET_NORMAL_DISPLAY  0xA6
+#define SH1122_SET_REVERSE_DISPLAY 0xA7
+#define SH1122_SET_ROW_ADDR        0xB0
+
+#define SH1122_SET_COLUMN_ADDR_HIGH(n)   0x10 + (n >> 4)
+#define SH1122_SET_COLUMN_ADDR_LOW(n)    0x0 + (n & 0xf)
+#define SH1122_SET_DISPLAY_START_LINE(n) 0x40 + (n & 0x3f)
+#define SH1122_SET_VSL(n)                0x30 + (n & 0xf)
```

Comment:
`SH1122_SET_COLUMN_ADDR_HIGH(x / 2)` expanded through `0x10 + (n >> 4)` becomes `0x10 + (x / 2 >> 4)` which [works](https://en.cppreference.com/w/c/language/operator_precedence), but having parenthesis around the whole expression and every argument always avoid subtle macro expansion bugs.

```suggestion
#define SH1122_SET_COLUMN_ADDR_HIGH(n)   (0x10 + ((n) >> 4))
#define SH1122_SET_COLUMN_ADDR_LOW(n)    (0x0 + ((n) & 0xf))
#define SH1122_SET_DISPLAY_START_LINE(n) (0x40 + ((n) & 0x3f))
#define SH1122_SET_VSL(n)                (0x30 + ((n) & 0xf))
```

Reply:
Done